pageserver/src/metrics.rs

```diff
@@ -520,10 +539,21 @@ pub struct TimelineMetrics {
     pub current_logical_size_gauge: UIntGauge,
     pub num_persistent_files_created: IntCounter,
     pub persistent_bytes_written: IntCounter,
+
+    // these 4 belong together
+    pub evictions: IntCounter,
+    evictions_with_low_residence_duration_data_source: &'static str,
+    evictions_with_low_residence_duration_threshold: Duration,
+    pub evictions_with_low_residence_duration: IntCounter,
```

Comment:
Too ugly? Do we want a proper abstraction for this?

Reply:
Perhaps the 3 trivially go together, but the `evictions` is just nice to have, should remain separate I think.

`ThresholdedEvictions` or just `AlertedEvictions`, or `TrashingEvictions`, `EvictedTooSoon`? :)

Alerted might be a good one since that is the goal here.

Reply:
With "proper abstraction", I meant something like a `ThresholdCounter` type that internalizes the aspects of
* only count if below / above a threshold
* attach the threshold label as a const label to the metric

But after some dabbling, I have come to the conclusion that the underlying `prometheus` crate isn't suitable for that. We'd need some kind of currying-ability for labels, and I don't want to spend any effort on that right now.

I'll give the grouping of the 3 related members a try though.

Reply:
Aah oki I wasn't think of generalization before we have 3, but I guess that'd be fine. Non-generalized would work well.